modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteHashFilter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.calcite.rel;
+
+import java.util.List;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.SingleRel;
+import org.apache.ignite.internal.processors.query.calcite.prepare.RelTarget;
+import org.apache.ignite.internal.processors.query.calcite.prepare.RelTargetAware;
+import org.apache.ignite.internal.processors.query.calcite.trait.DistributionTraitDef;
+
+import static org.apache.calcite.rel.RelDistribution.Type.BROADCAST_DISTRIBUTED;
+import static org.apache.calcite.rel.RelDistribution.Type.HASH_DISTRIBUTED;
+
+/**
+ *
+ */
+public class IgniteHashFilter extends SingleRel implements IgniteRel, RelTargetAware {
```

Comment:
IMO it is a poor name for this process. It can be easily confused with `IgniteFilter` despite it is a kind of exchange. I think we need to rename it to something more meaningful. For example in the [SQL Server PDW paper](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.473.9943&rep=rep1&type=pdf) this node is called "Trim move". What if we call it in the same manner: `TreamExchange`? Or simply `BroadcastToHashExchange`?

Reply:
You right the name is ugly, I tried to differ the rel node from Exchange since IgniteExchange is a special logical node, a cutting point for a splitter while IgniteHashFilter describes a physical operation. TrimExchange sounds better, it at least describes what actually is being done here.